eth/fetcher/block_fetcher.go

```diff
@@ -885,3 +890,23 @@ func (f *BlockFetcher) forgetBlock(hash common.Hash) {
 		delete(f.queued, hash)
 	}
 }
+
+func broadcastBlockEuclidV2(broadcastBlock blockBroadcasterFn, isEuclidV2 func(uint64) bool) blockBroadcasterFn {
+	return func(block *types.Block, propagate bool) {
+		block = block.CopyBlockDeepCopyHeader()
```

Comment:
Why copy the block here?

Reply:
Deprecated. But for completeness: this is the cost of dealing with modifying the header when sending it over the network: have to deep copy the header so that the sent header is modified without modified the internal one.